pkg/decision/composite_feature_service.go

```diff
@@ -44,14 +44,18 @@ func NewCompositeFeatureService(compositeExperimentService ExperimentService) *C
 
 // GetDecision returns a decision for the given feature and user context
 func (f CompositeFeatureService) GetDecision(decisionContext FeatureDecisionContext, userContext entities.UserContext) (FeatureDecision, error) {
-	for _, featureDecisionService := range f.featureServices {
+	lastElementIndex := len(f.featureServices) - 1
```

Comment:
There is a cleaner way to do this where you declare a FeatureDecision and an error var. You can then loop through and assign the result of calling `.GetDecision` to it and after the loop you can return whatever those variables were set to

Reply:
@mikeng13 , I think we need to return err all the time here. "no decision was made" error is not applicable anymore (we can log the warning), but our return statement will be always ```return FeatureDecision{}, err ``` ( err being the last iteration from the loop). is that correct ?

Reply:
Yes that is correct!

Reply:
@mikeng13  Requested changes made.